lib/Parser/Parse.cpp

```diff
@@ -1083,163 +985,340 @@ ParseNodePtr Parser::CreateSuperReferenceNode(OpCode nop, ParseNodePtr pnode1, P
     return pnode;
 }
 
-ParseNodePtr Parser::CreateTriNode(OpCode nop, ParseNodePtr pnode1,
-                                   ParseNodePtr pnode2, ParseNodePtr pnode3)
+ParseNodeProg * Parser::CreateProgNode(bool isModuleSource)
+{
+    ParseNodeProg * pnodeProg;
+
+    if (isModuleSource)
+    {
+        pnodeProg = CreateNodeForOpT<knopModule>();
+
+        // knopModule is not actually handled anywhere since we would need to handle it everywhere we could
+        // have knopProg and it would be treated exactly the same except for import/export statements.
+        // We are only using it as a way to get the correct size for PnModule.
+        // Consider: Should we add a flag to PnProg which is false but set to true in PnModule?
+        //           If we do, it can't be a virtual method since the parse nodes are all in a union.
```

Comment:
This comment isn't entirely accurate anymore. It is probably also worth revisiting this now that we could do a proper subclass of PnProg.

Reply:
Yep.   Will do that next.